### PR TITLE
Fixed vite vulnerability

### PR DESCRIPTION
**Pull request by Bayheck**
Fixed Vite xss vulnerability in demo-app by updating angular from version 16 to 17.
Also updated .github/workflows/test-e2e.yml from Node 16 to 18 because angular cli requires version 18.13 or higher. 

closes https://github.com/DevExpress/testcafe-private/issues/381

### CODE REVIEW

File: demo-app/angular.json [hunk 1]
```diff
@@ -64,24 +64,24 @@
         "serve": {
           "builder": "@angular-devkit/build-angular:dev-server",
           "options": {
-            "browserTarget": "demo-app:build",
             "host": "localhost",
-            "port": 4200
+            "port": 4200,
+            "buildTarget": "demo-app:build"
```
Comment (AlexKamaev) [inline demo-app/angular.json:69]: Just a formatting issue. Please do not change the order of these settings if possible. We do not want to highligh the `port: 4200` line as change.

Comment (AlexKamaev) [reply demo-app/angular.json:69]: Is renaming from "browserTarget" to "buildTarget" required? It's not a problem obviously, just interesting

File: demo-app/angular.json [hunk 2]
```diff
@@ -64,24 +64,24 @@
...
           "configurations": {
             "production": {
-              "browserTarget": "demo-app:build:production"
+              "buildTarget": "demo-app:build:production"
             },
             "stubbed": {
-              "browserTarget": "demo-app:build",
-              "publicHost": "localhost:4200"
+              "publicHost": "localhost:4200",
+              "buildTarget": "demo-app:build"
```
Comment (AlexKamaev) [inline demo-app/angular.json:77]: Please leave the initial settings order as it was

**Review by AlexKamaev — CHANGES_REQUESTED** [review]

File: demo-app/angular.json (hunk 1 above)
Comment (Bayheck) [reply demo-app/angular.json:69]: Thank you for review. "browserTarget" is now deprecated https://angular.io/cli/serve

Comment (Bayheck) [reply demo-app/angular.json:69]: It was automatically replaced when I ran "ng update"

**Review by Aleksey28 — APPROVED** [review]

**Review by AlexKamaev — APPROVED** [review]

**Pull request merged**